### PR TITLE
Simplifies LocationRuntimeModule binding (Part 1)

### PR DESCRIPTION
**Pull request by chtyim**
- Improves Locations.getLocationFromAbsolutePath and use it when appropriate
- Get rids of RootLocationFactory and favor for the
  getLocationFromAbsolutePath method instead as it is much cleaner

### CODE REVIEW

File: cdap-master/src/main/java/co/cask/cdap/master/startup/MasterStartupTool.java [hunk 1]
```diff
@@ -163,7 +165,13 @@ static Injector createInjector(CConfiguration cConf, Configuration hConf) {
         @Override
         protected void configure() {
           bind(FileContext.class).toProvider(FileContextProvider.class).in(Scopes.SINGLETON);
-          bind(LocationFactory.class).toProvider(RootLocationFactoryProvider.class).in(Scopes.SINGLETON);
+        }
+
+        @Provides
+        @Singleton
+        private LocationFactory providesLocationFactory(Configuration hConf, CConfiguration cConf, FileContext fc) {
```
Comment (chtyim) [inline cdap-master/src/main/java/co/cask/cdap/master/startup/MasterStartupTool.java:172]: This addition is temporary. Should be just using `LocationRuntimeModule`.getDistributedModule`. However it requires more refactoring of the location module (part II of the refactoring).

**Review by albertshau — APPROVED** [review]
lgtm

**Review by albertshau — APPROVED** [review]
lgtm

**Comment by chtyim** [comment]
Building http://builds.cask.co/browse/CDAP-DUT5548

**Pull request merged**